copy/copy.go

```diff
@@ -241,9 +241,20 @@ func copyLayers(manifestUpdates *types.ManifestUpdateOptions, dest types.ImageDe
 		cl, ok := copiedLayers[srcLayer.Digest]
 		if !ok {
 			fmt.Fprintf(reportWriter, "Copying blob %s\n", srcLayer.Digest)
-			destInfo, diffID, err := copyLayer(dest, rawSource, srcLayer, diffIDsAreNeeded, canModifyManifest, reportWriter)
-			if err != nil {
-				return err
+			var (
+				destInfo types.BlobInfo
+				diffID   string
+				err      error
+			)
+			if !dest.SupportsForeignLayers() && len(srcLayer.URLs) != 0 {
+				destInfo = srcLayer
+				// TODO: diffID?
```

Comment:
As it happens, DiffIDs are, at least currently, needed only when converting _from_ schema1, in which case `src.LayerInfos` will not have URLs because schema1 does not support them. So, luckily enough, we don’t need to be handling this; just add a check
```
if diffIDsAreNeeded {
   return errors.New("getting DiffIDs for foreign layers is unimplemented") // and a comment explaining schema1
}
```